### PR TITLE
Add `npm run reset` command to ensure a clean repo state after switching branches.

### PR DESCRIPTION
**Pull request by mpiroc**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Infrastructure (change which improves the lifecycle management (CI/CD, build, package, deploy, lint, etc) of the application, but does not change functionality.)
- [ ] Technical debt (change which improves the maintainability of the codebase, but does not change functionality)
- [ ] Testing (change which modifies or adds test coverage, but does not change functionality)
- [ ] Documentation (change which modifies or adds documentation, but does not change functionality)

## Description

Add an `npm run reset` command that:

* Cleans the `./out` directory.
* Clean the `./node_modules` directory.
* Runs `npm install`

## Motivation and Context

I often find myself manually running these commands after changing branches. Adding an npm command streamlines this process.

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: build-scripts/clean.js [hunk 1]
```diff
@@ -6,44 +6,57 @@
...
+const rmdir = util.promisify(fs.rmdir)
+const stat = util.promisify(fs.stat)
+const unlink = util.promisify(fs.unlink)
+
+// Recursive delete without requiring a third-party library. This allows the script
+// to be run before `npm install`.
+async function rdelete(path) {
+    const stats = await stat(path);
+    if (stats.isFile()) {
+        await unlink(path);
```
Comment (bryceitoc9) [inline build-scripts/clean.js]: `try/catch`?

File: build-scripts/clean.js [hunk 2]
```diff
@@ -6,44 +6,57 @@
...
+    } else if (stats.isDirectory()) {
+        const promises = (await readdir(path))
+            .map(child => rdelete(_path.join(path, child)));
+
+        await Promise.all(promises);
+        await rmdir(path);
```
Comment (bryceitoc9) [inline build-scripts/clean.js]: `try/catch`?

File: build-scripts/clean.js (hunk 1 above)
Comment (mpiroc) [reply build-scripts/clean.js]: If `unlink` or `rmdir` fail, we know that removal of the top-level directory will fail. Letting the error bubble up to the try/catch around the top-level directory prevents us from continuing to attempt an operation that we know will fail.

File: build-scripts/clean.js (hunk 2 above)
Comment (mpiroc) [reply build-scripts/clean.js]: See above.

**Pull request merged**